Heavy-Division/B78XH
# refactor: Organize repo

# Title
refactor: begin repo refactor





# Description 
The current repo structure isn't clear. This change will better organize the project structure into Heavy Source files, simulator build files, and external assets (fonts, etc.), and significantly reduce package size by separating sim build files from source files, and refcator/add some build scripts.


## Major changes:
 - Added the [B78XH-wasm]() repo as a git submodule in src/. 
 - All files required for simulator build are now located in src/base
 - localization files have been moved to src/localization/msfs
 - The root package.json now has script for the simulator build, ensuring only the necessary files are present in the installed/built aircraft, minimizing package size.
 - The wasm binary is now built from the root of the project. This change is dependent on merging changes to the wasm repository's main branch to work.
 

## Minor changes:
- Added changelog 
- Moved all licenses to project root
- Fixed typos and missing info in package.json
 
# Source material
N/A



# Testing Instructions 
Ensure the build scripts work:

1. Clone the repo
2. run `npm install` to fetch dependencies
3. run `npm run build:b78xh` 
4. check that the following files & directories are present in the out/hd-aircraft-b78x directory:
 - .locPak (localization) files
 - SimObjects
 - ModelBehaviorDefs
 - html_ui
 - SimObjects/Airplane/Heavy-Division-B78XH-Mod/panel/b78xh.wasm

take the `hd-aircraft-b78x` folder and place it in the Community folder. Ensure the mod is working normally.


# Test Results
![image](https://github.com/Heavy-Division/B78XH/assets/52870481/d53b3eb5-8cca-4d98-96c6-a61c3c6d3099)
![image](https://github.com/Heavy-Division/B78XH/assets/52870481/a8f67ecb-db1c-48c3-bfa4-8df5b03d740a)




# Discord Username
HowNowBrownCrow
